src/Common/src/Interop/Unix/System.Native/Interop.MessageHeader.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        internal unsafe struct IOVector
+        {
+            public byte* Base;
+            public UIntPtr Count;
```

Comment:
The layout of this type is intended to exactly match the layout of the platform's `struct iovec` in order to avoid expensive allocation + copying on each vectored receive or send. The layout of `struct iovec` is always
```
struct iovec {
   void* iov_base;
   size_t iov_len;
}
```

Hence the `UIntPtr`. I'll add a comment to the native definition (and I should point out that there are already asserts in this change to verify that the layouts of an `IOVector` and a `struct iovec` are identical).